syntax = "proto3";

package weather;

// 定義 WeatherService 服務
service WeatherService {
  rpc GetWeather (WeatherRequest) returns (WeatherResponse);
}

// 請求格式
message WeatherRequest {
  string country = 1;
}

// 回應格式
message WeatherResponse {
  int32 status = 1;
  string message = 2;
  repeated WeatherData result = 3;
}

// 天氣數據格式
message WeatherData {
  string date = 1;
  string morning = 2;
  string night = 3;
  string morning_temp = 4;
  string night_temp = 5;
}
